src/main/java/org/glassfish/concurro/cdi/lock/Lock.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package org.glassfish.concurro.cdi.lock;
```

Comment:
Wouldn't it be simpler to move this into the existing `org.glassfish.concurro.cdi` package instead of creating a new package? This is supposed to be used by application developers, so it's simpler to have it the `cdi` package, instead of another package (thinking about Javadoc where packages are displayed in different pages).

Reply:
Might be... I was thinking ahead for pooled (https://github.com/omnifaces/omniservices/tree/develop/src/main/java/org/omnifaces/services/pooled) which has a lot more classes.

Eventually (in EE 12) application developers would not use anything of concurro directly of course, but would use the EE 12 annotation only.

I'll move it to just CDI for now, and we'll take it from there.

Reply:
Maybe a compromise is to move the annotation to the `cdi` package, and keep the intercpetor in the `lock` package? With `@Pooled`, you could do the same. Then the API would be in the `cdi` package, and implementations in their own packages.